apps/emqx_resource/src/emqx_resource.erl

```diff
@@ -227,9 +228,26 @@ restart(InstId, Opts) ->
 stop(InstId) ->
     call_instance(InstId, {stop, InstId}).
 
+-spec health_check(instance_id(), timeout()) -> ok | {error, Reason :: term()}.
+health_check(InstId, Timeout) ->
+    try call_instance(InstId, {health_check, InstId}, Timeout)
+    catch 
+         _ -> _ = set_health_check_status_stoped(InstId),
```

Comment:
The `do_health_check/1` in emqx_resource_instance will set the status, so we don't need change the status again here.